File: fluttermultiplatform/lib/api/shipment_api.dart
import 'api_client.dart';

class PaymentApi {
  static Future createPayment(String orderId) => api.post('/payments/create', data: {'order_id': orderId});
  static Future getPayment(String orderId) => api.get('/payments/$orderId');
}

class ShipmentApi {
  static Future create(Map<String, dynamic> data) => api.post('/admin/shipments', data: data);
  static Future getMine() => api.get('/shipments');
  static Future getAll() => api.get('/admin/shipments');
  static Future updateStatus(String id, Map<String, dynamic> data) => api.put('/shipments/$id/status', data: data);
  static Future getLogs(String id) => api.get('/shipments/$id/logs');
}

class WithdrawalApi {
  static Future create(Map<String, dynamic> data) => api.post('/withdrawals', data: data);
  static Future getMine() => api.get('/withdrawals');
  static Future getAll() => api.get('/admin/withdrawals');
  static Future update(String id, Map<String, dynamic> data) => api.put('/admin/withdrawals/$id', data: data);
}

File: frontend/src/api/shipmentApi.ts
import api from './axios';
export const paymentApi = {
    createPayment: (orderId: string) => api.post('/payments/create', { order_id: orderId }),
    getPayment: (orderId: string) => api.get(`/payments/${orderId}`),
};
export const shipmentApi = {
    create: (data: Record<string, unknown>) => api.post('/admin/shipments', data),
    getMyShipments: () => api.get('/shipments'),
    getAllShipments: () => api.get('/admin/shipments'),
    updateStatus: (id: string, data: { status: string; location?: string; notes?: string }) =>
        api.put(`/shipments/${id}/status`, data),
    getLogs: (id: string) => api.get(`/shipments/${id}/logs`),
};
export const withdrawalApi = {
    create: (data: { amount: number; bank_name: string; account_number: string; account_name: string }) =>
        api.post('/withdrawals', data),
    getMyWithdrawals: () => api.get('/withdrawals'),
    getAllWithdrawals: () => api.get('/admin/withdrawals'),
    update: (id: string, data: { status: string; notes?: string }) =>
        api.put(`/admin/withdrawals/${id}`, data),
};
